### PR TITLE
refactor: Minor refactor in ingester lifecycle tests

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
I am studying the code for https://github.com/influxdata/influxdb_iox/issues/4928 and playing with it is a good way to learn; less code is better code

# Changes:
1. Remove `TestLifeCycleManager`

### CODE REVIEW

File: ingester/src/job.rs [hunk 1]
```diff
@@ -25,6 +25,7 @@ impl Job {
 /// The global job registry
 #[derive(Debug)]
 pub struct JobRegistry {
+    pub metric_registry: Arc<metric::Registry>,
```
Comment (tustvold) [inline ingester/src/job.rs:28]: Why does this need to be pub?

File: iox_time/src/lib.rs [hunk 2]
```diff
@@ -182,6 +182,9 @@ impl Time {
 }
 
 pub trait TimeProvider: std::fmt::Debug + Send + Sync + 'static {
+    /// Returns `self` as [`std::any::Any`] for runtime downcasts
+    fn as_any(&self) -> &dyn std::any::Any;
```
Comment (tustvold) [inline iox_time/src/lib.rs:186]: This is only needed for tests, and so it seems a little strange to have it leak out in this way...

**Review by tustvold — COMMENTED** [review]
I'm not sure I agree with this change, it effectively seems to leak test harness code into non-test code, which seems undesirable...

**Comment by alamb** [comment]
not worth pursuing

**Pull request closed without merging**